internal/konnect/client.go

```diff
@@ -0,0 +1,169 @@
+package konnect
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/adminapi"
+	tlsutil "github.com/kong/kubernetes-ingress-controller/v2/internal/util/tls"
+)
+
+type KonnectAdminClient struct {
```

Comment:
Just as the golangci-lint annotation says: better rename this to `AdminClient`

Reply:
done